src/Init/Data/Char/Basic.lean

```diff
@@ -15,6 +16,9 @@ Note that values in `[0xd800, 0xdfff]` are reserved for [UTF-16 surrogate pairs]
 
 namespace Char
 
+/-- Maximum valid Unicode codepoint -/
+def max : Nat := 1114111
```

Comment:
This seems to be unused. In any case, I think `Char.max` should be of type `Char`. Also, declarations with a generic name like `max` should be marked as `protected` so that saying `open Char` won't make `Char.max` accessible as just `max` (which would be very confusing).